src/huggingface_hub/hf_api.py

```diff
@@ -606,9 +618,66 @@ def list_models(
         d = r.json()
         return [ModelInfo(**x) for x in d]
 
+    def _unpack_model_filter(self, model_filter: ModelFilter):
+        """
+        Unpacks a `ModelFilter` into something readable for `list_models`
+        """
+        model_str = ""
+        tags = []
+
+        # Handling author
+        if model_filter.author is not None:
+            model_str = f"{model_filter.author}/"
+
+        # Handling model_name
+        if model_filter.model_name is not None:
+            model_str += model_filter.model_name
+
+        filter_tuple = []
+
+        # Handling tasks
+        if model_filter.task is not None:
+            if not isinstance(model_filter.task, (list, tuple)):
+                model_filter.task = [model_filter.task]
+            for task in model_filter.task:
+                filter_tuple.append(task)
+
+        # Handling dataset
+        if model_filter.trained_dataset is not None:
+            if not isinstance(model_filter.trained_dataset, (list, tuple)):
+                model_filter.trained_dataset = [model_filter.trained_dataset]
+            for dataset in model_filter.trained_dataset:
+                if "dataset:" not in dataset:
+                    dataset = f"dataset:{dataset}"
+                filter_tuple.append(dataset)
+
+        # Handling framework
+        if model_filter.framework:
+            if not isinstance(model_filter.framework, (list, tuple)):
+                model_filter.framework = [model_filter.framework]
+            for framework in model_filter.framework:
+                filter_tuple.append(framework)
```

Comment:
Could this be `filter_tuple.extend(framework)` instead?

Reply:
See note here https://github.com/huggingface/huggingface_hub/pull/553#discussion_r779558602